src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionSet2.cs

```diff
@@ -144,5 +142,54 @@ public override void Recalculate()
         {
             // Do nothing.  Our controller will already recalculate if necessary.
         }
+
+        private CompletionRules GetCompletionRules()
+        {
+            foregroundObject.AssertIsForeground();
+            if (_completionRules == null)
+            {
+                var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                if (document != null)
+                {
+                    var service = document.Project.LanguageServices.GetService<ICompletionService>();
+                    if (service != null)
+                    {
+                        _completionRules = service.GetCompletionRules();
+                    }
+                }
+            }
+
+            return _completionRules;
+        }
+
+#if false
+        public override IReadOnlyList<Span> GetHighlightedSpansInDisplayText(string displayText)
+#else
+        public IReadOnlyList<Span> GetHighlightedSpansInDisplayText(string displayText)
+#endif
+        {
+            var rules = this.GetCompletionRules();
+            if (rules != null)
+            {
+                var status = this.SelectionStatus;
+                var vsCompletion = status.Completion;
+                if (vsCompletion != null)
+                {
+                    var completionItem = GetCompletionItem(vsCompletion);
+                    if (completionItem != null)
+                    {
+                        var textSnapshot = _subjectBuffer.CurrentSnapshot;
+                        var filterText = textSnapshot.GetText(completionItem.FilterSpan.ToSpan());
+                        var highlightedSpans = rules.GetHighlightedSpans(completionItem, filterText);
+                        if (highlightedSpans != null)
+                        {
+                            return highlightedSpans.Select(s => s.ToSpan()).ToArray();
```

Comment:
My desperate hope is that the editor is only calling this for the items in view.  Hence it would be a couple of dozen of calls at most.  In which case i don't care :)